metadatastore/commands.py

```diff
@@ -39,167 +99,512 @@ def inner(*args, **kwargs):
+        EventDescriptor document
+
+    data_table : DataFrame
+        All of the data as a DataFrame indexed on sequence number
+
+    uids : Series
+        The uids of each of the events as a `Series` indexed on sequence number
+
+    times : Series
+        The times of each event as a `Series` indexed on sequence number
+
+    uids : Series
+        The uids of each event as a `Series` indexed on sequence number
+
+    timestamps_table : DataFrame
+        The timestamps of each of the measurements as a `DataFrame`.  Same
+        columns as `data_table` indexed on sequence number
+    """
+    import pandas as pd
+    desc_uid = doc_or_uid_to_uid(descriptor)
+    descriptor = descriptor_given_uid(desc_uid)
+    # this will get more complicated once transpose caching layer is in place
+    all_events = list(fetch_events_generator(desc_uid))
+
+    # get event sequence numbers
+    seq_num = pd.Series([ev['seq_num'] for ev in all_events])
+
+    # get event times
+    times = pd.to_datetime([ev['time'] for ev in all_events], unit='s')
+    times.index = seq_num
+
+    # get uids
+    uids = pd.Series([ev['uid'] for ev in all_events], index=seq_num)
+
+    # get data values
+    data_table = pd.DataFrame([ev['data'] for ev in all_events], index=seq_num)
+
+    # get timestamps
+    ts = pd.DataFrame([ev['timestamps'] for ev in all_events], index=seq_num)
+    timestamps_table = pd.DataFrame({k: pd.to_datetime(ts[k], unit='s')
+                                     for k in ts})
+    # return the whole lot
+    return descriptor, data_table, times, uids, timestamps_table
+
+
+def db_disconnect():
+    """Helper function to deal with stateful connections to mongoengine"""
+    mongoengine.connection.disconnect(ALIAS)
+    for collection in [RunStart, RunStop, EventDescriptor,
+                       Event, DataKey]:
+        collection._collection = None
+
+
+def db_connect(database, host, port):
+    """Helper function to deal with stateful connections to mongoengine
+
+    .. warning
+
+       This will silently ignore input if the database is already
+       connected, even if the input database, host, or port are
+       different than currently connected.  To change the database
+       connection you must call `db_disconnect` before attempting to
+       re-connect.
     """
-    return {key: [data_dict['value'], data_dict['timestamp']]
-            for key, data_dict in six.iteritems(event_dict)}
+    return connect(db=database, host=host, port=port, alias=ALIAS)
 
 
 # database INSERTION ###################################################
 
 @_ensure_connection
-def insert_run_start(time, scan_id, beamline_id, beamline_config, uid=None,
-                     owner=None, group=None, project=None, custom=None):
-    """Provide a head for a sequence of events. Entry point for an
-    experiment's run.
+def insert_run_start(time, scan_id, beamline_id, uid,
+                     owner='', group='', project='', custom=None):
```

Comment:
Because the `None` was just being converted to `''` below.  Strings are immutable so they can be used as default arguments where as dicts are mutable and can not be.